### PR TITLE
478 migration with attribute refs causes error

### PR DESCRIPTION
**Pull request by yflim**
#### SUMMARY
Fix exception thrown on importing a database export containing attribute refs #478

##### Bugfix
- [x] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Architecture Decision Record added if design changes necessary
- [x] Formatting checked
- [ ] `CHANGELOG.md` updated

### CODE REVIEW

**Comment by yflim** [comment]
Note that most of the commits here are from branch [477-history-migration-not-working](https://github.com/replikativ/datahike/tree/477-history-migration-not-working) for [PR 496](https://github.com/replikativ/datahike/pull/496). The only one strictly for this issue/PR is the last, [743db12](https://github.com/replikativ/datahike/commit/743db12fffd6f529967fc558b9b0ae4ad1f7a198).

**Comment by yflim** [comment]
Not sure what "This can happen as long as we are arriving at version 1.0.0 and will always be communicated through the Changelog" under [Migration & Backup](This can happen as long as we are arriving at version 1.0.0 and will always be communicated through the Changelog) in the README means, so left it alone even if it should be edited.

**Comment by jsmassa** [comment]
Should be rebased after #496 has been merged to make actual changes more visible

**Review by kordano — CHANGES_REQUESTED** [review]

File: src/datahike/migrate.clj [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
-  (cbor/spit-all path (map seq (sort-by
-                                (juxt d/datom-tx :e)
-                                (api/datoms (if (db/-keep-history? db) (api/history db) db) :eavt)))))
+  [conn path]
+  (let [db @conn
+        cfg (:config db)]
+    (cbor/spit-all path (cond->> (sort-by
+                                  (juxt d/datom-tx :e)
+                                  (api/datoms (if (:keep-history? cfg) (api/history db) db) :eavt))
+                          (:attribute-refs? cfg) (remove #(= (d/datom-tx %) c/tx0))
```
Comment (jsmassa) [inline src/datahike/migrate.clj:16]: What happens if the system entities are changed between export and import datahike versions? For example when the order changes or there are more or less system entities than before?

Comment (yflim) [reply src/datahike/migrate.clj:16]: I assumed any changes to system entities would only be additive. I guess that isn't a sufficiently realistic or safe assumption? Also, as I understand, the migration code here isn't a permanent solution—a role Wanderung is supposed to fulfil at some point. Please let me know.

Comment (jsmassa) [reply src/datahike/migrate.clj:16]: I guess, if it is only a temporary solution, it should be ok for now, but maybe put at least a note in there about the assumption. 
I was of the impression, the system entities are in the database for exactly that reason - so the case of invalid references would never occur. But I wouldn't be surprised if none of our functions support importing the schema and I'd say that would be my mistake, sorry. I will open an issue or discussion about it to not lose track of the subject.

**Review by kordano — APPROVED** [review]

**Pull request merged**